Zend/zend_language_parser.y

```diff
@@ -1075,7 +1102,13 @@ new_variable:
 ;
 
 member_name:
-		T_STRING		{ $$ = $1; }
+		identifier { $$ = $1; }
+	|	'{' expr '}'	{ $$ = $2; }
+	|	simple_variable	{ $$ = zend_ast_create(ZEND_AST_VAR, $1); }
+;
+
+property_name:
```

Comment:
Because property names have sigils and therefore no naming limitations, if you handle it the same way as the other member names they will be restricted by the inclusive `SEMI_RESERVED` rule https://github.com/php/php-src/pull/1158/files#diff-7eff82c2c5b45db512a9dc49fb990bb8R273

I'm sure you don't want that.